charms/katib-db-manager/src/charm.py

```diff
@@ -136,17 +158,132 @@ def _check_leader(self):
             self.logger.info("Not a leader, skipping setup")
             raise ErrorWithStatus("Waiting for leadership", WaitingStatus)
 
-    def _check_mysql(self):
+    def _on_mysql_relation(self, event):
+        """Check for existing database relations and process mysql relation if needed."""
+        # check for relational-db relation
+        # relying on KeyError to ensure that relational-db relation is not present
+        try:
+            relation = self.model.get_relation("relational-db")
+            if relation:
+                self.logger.warnning(
+                    "Failed to create mysql relation due to existing relational-db relation."
+                )
+                return
+        except KeyError:
+            # relational-db relation was not found, proceed
```

Comment:
no, we need to add warning and juju status telling that mysql is deprecated

Reply:
Warning will go into log. Status will be overwritten pretty quickly, once we start processing mysql relation data.